public final class kotlinx.coroutines.intrinsics.UndispatchedKt extends java.lang.Object
{

    public static final void startCoroutineUndispatched(kotlin.jvm.functions.Function1, kotlin.coroutines.Continuation)
    {
        kotlin.jvm.functions.Function1 $r0;
        kotlin.coroutines.Continuation $r1, $r2;
        kotlin.coroutines.CoroutineContext $r3;
        java.lang.Object $r4, $r5;
        kotlin.Result$Companion $r6;
        java.lang.Throwable $r7, $r9;
        java.lang.NullPointerException $r8;

        $r0 := @parameter0: kotlin.jvm.functions.Function1;

        $r1 := @parameter1: kotlin.coroutines.Continuation;

        $r2 = staticinvoke <kotlin.coroutines.jvm.internal.DebugProbesKt: kotlin.coroutines.Continuation probeCoroutineCreated(kotlin.coroutines.Continuation)>($r1);

     label01:
        $r3 = interfaceinvoke $r1.<kotlin.coroutines.Continuation: kotlin.coroutines.CoroutineContext getContext()>();

        $r4 = staticinvoke <kotlinx.coroutines.internal.ThreadContextKt: java.lang.Object updateThreadContext(kotlin.coroutines.CoroutineContext,java.lang.Object)>($r3, null);

     label02:
        if $r0 == null goto label07;

     label03:
        $r5 = staticinvoke <kotlin.jvm.internal.TypeIntrinsics: java.lang.Object beforeCheckcastToFunctionOfArity(java.lang.Object,int)>($r0, 1);

        $r0 = (kotlin.jvm.functions.Function1) $r5;

        $r5 = interfaceinvoke $r0.<kotlin.jvm.functions.Function1: java.lang.Object invoke(java.lang.Object)>($r2);

     label04:
        staticinvoke <kotlinx.coroutines.internal.ThreadContextKt: void restoreThreadContext(kotlin.coroutines.CoroutineContext,java.lang.Object)>($r3, $r4);

     label05:
        $r4 = staticinvoke <kotlin.coroutines.intrinsics.IntrinsicsKt: java.lang.Object getCOROUTINE_SUSPENDED()>();

        if $r5 == $r4 goto label10;

        $r6 = <kotlin.Result: kotlin.Result$Companion Companion>;

        $r4 = staticinvoke <kotlin.Result: java.lang.Object constructor-impl(java.lang.Object)>($r5);

        interfaceinvoke $r2.<kotlin.coroutines.Continuation: void resumeWith(java.lang.Object)>($r4);

        return;

     label06:
        $r7 := @caughtexception;

        goto label08;

     label07:
        $r8 = new java.lang.NullPointerException;

        specialinvoke $r8.<java.lang.NullPointerException: void <init>(java.lang.String)>("null cannot be cast to non-null type (kotlin.coroutines.Continuation<T>) -> kotlin.Any?");

        throw $r8;

     label08:
        staticinvoke <kotlinx.coroutines.internal.ThreadContextKt: void restoreThreadContext(kotlin.coroutines.CoroutineContext,java.lang.Object)>($r3, $r4);

        throw $r7;

     label09:
        $r9 := @caughtexception;

        $r6 = <kotlin.Result: kotlin.Result$Companion Companion>;

        $r4 = staticinvoke <kotlin.ResultKt: java.lang.Object createFailure(java.lang.Throwable)>($r9);

        $r4 = staticinvoke <kotlin.Result: java.lang.Object constructor-impl(java.lang.Object)>($r4);

        interfaceinvoke $r2.<kotlin.coroutines.Continuation: void resumeWith(java.lang.Object)>($r4);

     label10:
        return;

        catch java.lang.Throwable from label01 to label02 with label09;
        catch java.lang.Throwable from label03 to label04 with label06;
        catch java.lang.Throwable from label04 to label05 with label09;
        catch java.lang.Throwable from label07 to label08 with label06;
        catch java.lang.Throwable from label08 to label09 with label09;
    }

    public static final void startCoroutineUndispatched(kotlin.jvm.functions.Function2, java.lang.Object, kotlin.coroutines.Continuation)
    {
        kotlin.jvm.functions.Function2 $r0;
        java.lang.Object $r1, $r5, $r6;
        kotlin.coroutines.Continuation $r2, $r3;
        kotlin.coroutines.CoroutineContext $r4;
        kotlin.Result$Companion $r7;
        java.lang.Throwable $r8, $r10;
        java.lang.NullPointerException $r9;

        $r0 := @parameter0: kotlin.jvm.functions.Function2;

        $r1 := @parameter1: java.lang.Object;

        $r2 := @parameter2: kotlin.coroutines.Continuation;

        $r3 = staticinvoke <kotlin.coroutines.jvm.internal.DebugProbesKt: kotlin.coroutines.Continuation probeCoroutineCreated(kotlin.coroutines.Continuation)>($r2);

     label01:
        $r4 = interfaceinvoke $r2.<kotlin.coroutines.Continuation: kotlin.coroutines.CoroutineContext getContext()>();

        $r5 = staticinvoke <kotlinx.coroutines.internal.ThreadContextKt: java.lang.Object updateThreadContext(kotlin.coroutines.CoroutineContext,java.lang.Object)>($r4, null);

     label02:
        if $r0 == null goto label07;

     label03:
        $r6 = staticinvoke <kotlin.jvm.internal.TypeIntrinsics: java.lang.Object beforeCheckcastToFunctionOfArity(java.lang.Object,int)>($r0, 2);

        $r0 = (kotlin.jvm.functions.Function2) $r6;

        $r1 = interfaceinvoke $r0.<kotlin.jvm.functions.Function2: java.lang.Object invoke(java.lang.Object,java.lang.Object)>($r1, $r3);

     label04:
        staticinvoke <kotlinx.coroutines.internal.ThreadContextKt: void restoreThreadContext(kotlin.coroutines.CoroutineContext,java.lang.Object)>($r4, $r5);

     label05:
        $r5 = staticinvoke <kotlin.coroutines.intrinsics.IntrinsicsKt: java.lang.Object getCOROUTINE_SUSPENDED()>();

        if $r1 == $r5 goto label10;

        $r7 = <kotlin.Result: kotlin.Result$Companion Companion>;

        $r1 = staticinvoke <kotlin.Result: java.lang.Object constructor-impl(java.lang.Object)>($r1);

        interfaceinvoke $r3.<kotlin.coroutines.Continuation: void resumeWith(java.lang.Object)>($r1);

        return;

     label06:
        $r8 := @caughtexception;

        goto label08;

     label07:
        $r9 = new java.lang.NullPointerException;

        specialinvoke $r9.<java.lang.NullPointerException: void <init>(java.lang.String)>("null cannot be cast to non-null type (R, kotlin.coroutines.Continuation<T>) -> kotlin.Any?");

        throw $r9;

     label08:
        staticinvoke <kotlinx.coroutines.internal.ThreadContextKt: void restoreThreadContext(kotlin.coroutines.CoroutineContext,java.lang.Object)>($r4, $r5);

        throw $r8;

     label09:
        $r10 := @caughtexception;

        $r7 = <kotlin.Result: kotlin.Result$Companion Companion>;

        $r1 = staticinvoke <kotlin.ResultKt: java.lang.Object createFailure(java.lang.Throwable)>($r10);

        $r1 = staticinvoke <kotlin.Result: java.lang.Object constructor-impl(java.lang.Object)>($r1);

        interfaceinvoke $r3.<kotlin.coroutines.Continuation: void resumeWith(java.lang.Object)>($r1);

     label10:
        return;

        catch java.lang.Throwable from label01 to label02 with label09;
        catch java.lang.Throwable from label03 to label04 with label06;
        catch java.lang.Throwable from label04 to label05 with label09;
        catch java.lang.Throwable from label07 to label08 with label06;
        catch java.lang.Throwable from label08 to label09 with label09;
    }

    public static final void startCoroutineUnintercepted(kotlin.jvm.functions.Function1, kotlin.coroutines.Continuation)
    {
        kotlin.jvm.functions.Function1 $r0;
        kotlin.coroutines.Continuation $r1;
        java.lang.Object $r2, $r3;
        kotlin.Result$Companion $r4;
        java.lang.Throwable $r5;
        java.lang.NullPointerException $r6;

        $r0 := @parameter0: kotlin.jvm.functions.Function1;

        $r1 := @parameter1: kotlin.coroutines.Continuation;

        $r1 = staticinvoke <kotlin.coroutines.jvm.internal.DebugProbesKt: kotlin.coroutines.Continuation probeCoroutineCreated(kotlin.coroutines.Continuation)>($r1);

        if $r0 == null goto label4;

     label1:
        $r2 = staticinvoke <kotlin.jvm.internal.TypeIntrinsics: java.lang.Object beforeCheckcastToFunctionOfArity(java.lang.Object,int)>($r0, 1);

        $r0 = (kotlin.jvm.functions.Function1) $r2;

        $r2 = interfaceinvoke $r0.<kotlin.jvm.functions.Function1: java.lang.Object invoke(java.lang.Object)>($r1);

     label2:
        $r3 = staticinvoke <kotlin.coroutines.intrinsics.IntrinsicsKt: java.lang.Object getCOROUTINE_SUSPENDED()>();

        if $r2 == $r3 goto label6;

        $r4 = <kotlin.Result: kotlin.Result$Companion Companion>;

        $r2 = staticinvoke <kotlin.Result: java.lang.Object constructor-impl(java.lang.Object)>($r2);

        interfaceinvoke $r1.<kotlin.coroutines.Continuation: void resumeWith(java.lang.Object)>($r2);

        return;

     label3:
        $r5 := @caughtexception;

        goto label5;

     label4:
        $r6 = new java.lang.NullPointerException;

        specialinvoke $r6.<java.lang.NullPointerException: void <init>(java.lang.String)>("null cannot be cast to non-null type (kotlin.coroutines.Continuation<T>) -> kotlin.Any?");

        throw $r6;

     label5:
        $r4 = <kotlin.Result: kotlin.Result$Companion Companion>;

        $r2 = staticinvoke <kotlin.ResultKt: java.lang.Object createFailure(java.lang.Throwable)>($r5);

        $r2 = staticinvoke <kotlin.Result: java.lang.Object constructor-impl(java.lang.Object)>($r2);

        interfaceinvoke $r1.<kotlin.coroutines.Continuation: void resumeWith(java.lang.Object)>($r2);

     label6:
        return;

        catch java.lang.Throwable from label1 to label2 with label3;
        catch java.lang.Throwable from label4 to label5 with label3;
    }

    public static final void startCoroutineUnintercepted(kotlin.jvm.functions.Function2, java.lang.Object, kotlin.coroutines.Continuation)
    {
        kotlin.jvm.functions.Function2 $r0;
        java.lang.Object $r1, $r3;
        kotlin.coroutines.Continuation $r2;
        kotlin.Result$Companion $r4;
        java.lang.Throwable $r5;
        java.lang.NullPointerException $r6;

        $r0 := @parameter0: kotlin.jvm.functions.Function2;

        $r1 := @parameter1: java.lang.Object;

        $r2 := @parameter2: kotlin.coroutines.Continuation;

        $r2 = staticinvoke <kotlin.coroutines.jvm.internal.DebugProbesKt: kotlin.coroutines.Continuation probeCoroutineCreated(kotlin.coroutines.Continuation)>($r2);

        if $r0 == null goto label4;

     label1:
        $r3 = staticinvoke <kotlin.jvm.internal.TypeIntrinsics: java.lang.Object beforeCheckcastToFunctionOfArity(java.lang.Object,int)>($r0, 2);

        $r0 = (kotlin.jvm.functions.Function2) $r3;

        $r1 = interfaceinvoke $r0.<kotlin.jvm.functions.Function2: java.lang.Object invoke(java.lang.Object,java.lang.Object)>($r1, $r2);

     label2:
        $r3 = staticinvoke <kotlin.coroutines.intrinsics.IntrinsicsKt: java.lang.Object getCOROUTINE_SUSPENDED()>();

        if $r1 == $r3 goto label6;

        $r4 = <kotlin.Result: kotlin.Result$Companion Companion>;

        $r1 = staticinvoke <kotlin.Result: java.lang.Object constructor-impl(java.lang.Object)>($r1);

        interfaceinvoke $r2.<kotlin.coroutines.Continuation: void resumeWith(java.lang.Object)>($r1);

        return;

     label3:
        $r5 := @caughtexception;

        goto label5;

     label4:
        $r6 = new java.lang.NullPointerException;

        specialinvoke $r6.<java.lang.NullPointerException: void <init>(java.lang.String)>("null cannot be cast to non-null type (R, kotlin.coroutines.Continuation<T>) -> kotlin.Any?");

        throw $r6;

     label5:
        $r4 = <kotlin.Result: kotlin.Result$Companion Companion>;

        $r1 = staticinvoke <kotlin.ResultKt: java.lang.Object createFailure(java.lang.Throwable)>($r5);

        $r1 = staticinvoke <kotlin.Result: java.lang.Object constructor-impl(java.lang.Object)>($r1);

        interfaceinvoke $r2.<kotlin.coroutines.Continuation: void resumeWith(java.lang.Object)>($r1);

     label6:
        return;

        catch java.lang.Throwable from label1 to label2 with label3;
        catch java.lang.Throwable from label4 to label5 with label3;
    }

    private static final void startDirect(kotlin.coroutines.Continuation, kotlin.jvm.functions.Function1)
    {
        kotlin.coroutines.Continuation $r0;
        kotlin.jvm.functions.Function1 $r1;
        java.lang.Object $r2, $r3;
        kotlin.Result$Companion $r4;
        java.lang.Throwable $r5;

        $r0 := @parameter0: kotlin.coroutines.Continuation;

        $r1 := @parameter1: kotlin.jvm.functions.Function1;

        $r0 = staticinvoke <kotlin.coroutines.jvm.internal.DebugProbesKt: kotlin.coroutines.Continuation probeCoroutineCreated(kotlin.coroutines.Continuation)>($r0);

     label1:
        $r2 = interfaceinvoke $r1.<kotlin.jvm.functions.Function1: java.lang.Object invoke(java.lang.Object)>($r0);

     label2:
        $r3 = staticinvoke <kotlin.coroutines.intrinsics.IntrinsicsKt: java.lang.Object getCOROUTINE_SUSPENDED()>();

        if $r2 == $r3 goto label4;

        $r4 = <kotlin.Result: kotlin.Result$Companion Companion>;

        $r2 = staticinvoke <kotlin.Result: java.lang.Object constructor-impl(java.lang.Object)>($r2);

        interfaceinvoke $r0.<kotlin.coroutines.Continuation: void resumeWith(java.lang.Object)>($r2);

        return;

     label3:
        $r5 := @caughtexception;

        $r4 = <kotlin.Result: kotlin.Result$Companion Companion>;

        $r2 = staticinvoke <kotlin.ResultKt: java.lang.Object createFailure(java.lang.Throwable)>($r5);

        $r2 = staticinvoke <kotlin.Result: java.lang.Object constructor-impl(java.lang.Object)>($r2);

        interfaceinvoke $r0.<kotlin.coroutines.Continuation: void resumeWith(java.lang.Object)>($r2);

     label4:
        return;

        catch java.lang.Throwable from label1 to label2 with label3;
    }

    public static final java.lang.Object startUndispatchedOrReturn(kotlinx.coroutines.internal.ScopeCoroutine, java.lang.Object, kotlin.jvm.functions.Function2)
    {
        kotlinx.coroutines.internal.ScopeCoroutine $r0;
        java.lang.Object $r1, $r3;
        kotlin.jvm.functions.Function2 $r2;
        kotlin.coroutines.Continuation $r4;
        java.lang.Throwable $r5;
        java.lang.NullPointerException $r6;
        kotlinx.coroutines.CompletedExceptionally $r7;
        kotlinx.coroutines.internal.Symbol $r8;
        boolean $z0;
        kotlin.coroutines.jvm.internal.CoroutineStackFrame $r9;

        $r0 := @parameter0: kotlinx.coroutines.internal.ScopeCoroutine;

        $r1 := @parameter1: java.lang.Object;

        $r2 := @parameter2: kotlin.jvm.functions.Function2;

        if $r2 == null goto label04;

     label01:
        $r3 = staticinvoke <kotlin.jvm.internal.TypeIntrinsics: java.lang.Object beforeCheckcastToFunctionOfArity(java.lang.Object,int)>($r2, 2);

        $r2 = (kotlin.jvm.functions.Function2) $r3;

        $r4 = (kotlin.coroutines.Continuation) $r0;

        $r1 = interfaceinvoke $r2.<kotlin.jvm.functions.Function2: java.lang.Object invoke(java.lang.Object,java.lang.Object)>($r1, $r4);

     label02:
        goto label06;

     label03:
        $r5 := @caughtexception;

        goto label05;

     label04:
        $r6 = new java.lang.NullPointerException;

        specialinvoke $r6.<java.lang.NullPointerException: void <init>(java.lang.String)>("null cannot be cast to non-null type (R, kotlin.coroutines.Continuation<T>) -> kotlin.Any?");

        throw $r6;

     label05:
        $r7 = new kotlinx.coroutines.CompletedExceptionally;

        specialinvoke $r7.<kotlinx.coroutines.CompletedExceptionally: void <init>(java.lang.Throwable,boolean,int,kotlin.jvm.internal.DefaultConstructorMarker)>($r5, 0, 2, null);

        $r1 = $r7;

     label06:
        $r3 = staticinvoke <kotlin.coroutines.intrinsics.IntrinsicsKt: java.lang.Object getCOROUTINE_SUSPENDED()>();

        if $r1 != $r3 goto label07;

        $r1 = staticinvoke <kotlin.coroutines.intrinsics.IntrinsicsKt: java.lang.Object getCOROUTINE_SUSPENDED()>();

        return $r1;

     label07:
        $r1 = virtualinvoke $r0.<kotlinx.coroutines.internal.ScopeCoroutine: java.lang.Object makeCompletingOnce$kotlinx_coroutines_core(java.lang.Object)>($r1);

        $r8 = <kotlinx.coroutines.JobSupportKt: kotlinx.coroutines.internal.Symbol COMPLETING_WAITING_CHILDREN>;

        if $r1 != $r8 goto label08;

        $r1 = staticinvoke <kotlin.coroutines.intrinsics.IntrinsicsKt: java.lang.Object getCOROUTINE_SUSPENDED()>();

        return $r1;

     label08:
        $z0 = $r1 instanceof kotlinx.coroutines.CompletedExceptionally;

        if $z0 == 0 goto label11;

        $r7 = (kotlinx.coroutines.CompletedExceptionally) $r1;

        $r5 = $r7.<kotlinx.coroutines.CompletedExceptionally: java.lang.Throwable cause>;

        $r5 = $r7.<kotlinx.coroutines.CompletedExceptionally: java.lang.Throwable cause>;

        $r4 = $r0.<kotlinx.coroutines.internal.ScopeCoroutine: kotlin.coroutines.Continuation uCont>;

        $z0 = staticinvoke <kotlinx.coroutines.DebugKt: boolean getRECOVER_STACK_TRACES()>();

        if $z0 == 0 goto label10;

        $z0 = $r4 instanceof kotlin.coroutines.jvm.internal.CoroutineStackFrame;

        if $z0 != 0 goto label09;

        goto label10;

     label09:
        $r9 = (kotlin.coroutines.jvm.internal.CoroutineStackFrame) $r4;

        $r5 = staticinvoke <kotlinx.coroutines.internal.StackTraceRecoveryKt: java.lang.Throwable access$recoverFromStackFrame(java.lang.Throwable,kotlin.coroutines.jvm.internal.CoroutineStackFrame)>($r5, $r9);

     label10:
        throw $r5;

     label11:
        $r1 = staticinvoke <kotlinx.coroutines.JobSupportKt: java.lang.Object unboxState(java.lang.Object)>($r1);

        return $r1;

        catch java.lang.Throwable from label01 to label02 with label03;
        catch java.lang.Throwable from label04 to label05 with label03;
    }

    public static final java.lang.Object startUndispatchedOrReturnIgnoreTimeout(kotlinx.coroutines.internal.ScopeCoroutine, java.lang.Object, kotlin.jvm.functions.Function2)
    {
        kotlinx.coroutines.internal.ScopeCoroutine $r0;
        java.lang.Object $r1, $r3;
        kotlin.jvm.functions.Function2 $r2;
        boolean $z0, $z1;
        kotlin.coroutines.Continuation $r4;
        java.lang.Throwable $r5;
        java.lang.NullPointerException $r6;
        kotlinx.coroutines.CompletedExceptionally $r7;
        kotlinx.coroutines.internal.Symbol $r8;
        kotlinx.coroutines.TimeoutCancellationException $r9;
        kotlinx.coroutines.Job $r10;
        kotlin.coroutines.jvm.internal.CoroutineStackFrame $r11;

        $r0 := @parameter0: kotlinx.coroutines.internal.ScopeCoroutine;

        $r1 := @parameter1: java.lang.Object;

        $r2 := @parameter2: kotlin.jvm.functions.Function2;

        $z0 = 0;

        if $r2 == null goto label04;

     label01:
        $r3 = staticinvoke <kotlin.jvm.internal.TypeIntrinsics: java.lang.Object beforeCheckcastToFunctionOfArity(java.lang.Object,int)>($r2, 2);

        $r2 = (kotlin.jvm.functions.Function2) $r3;

        $r4 = (kotlin.coroutines.Continuation) $r0;

        $r1 = interfaceinvoke $r2.<kotlin.jvm.functions.Function2: java.lang.Object invoke(java.lang.Object,java.lang.Object)>($r1, $r4);

     label02:
        goto label06;

     label03:
        $r5 := @caughtexception;

        goto label05;

     label04:
        $r6 = new java.lang.NullPointerException;

        specialinvoke $r6.<java.lang.NullPointerException: void <init>(java.lang.String)>("null cannot be cast to non-null type (R, kotlin.coroutines.Continuation<T>) -> kotlin.Any?");

        throw $r6;

     label05:
        $r7 = new kotlinx.coroutines.CompletedExceptionally;

        specialinvoke $r7.<kotlinx.coroutines.CompletedExceptionally: void <init>(java.lang.Throwable,boolean,int,kotlin.jvm.internal.DefaultConstructorMarker)>($r5, 0, 2, null);

        $r1 = $r7;

     label06:
        $r3 = staticinvoke <kotlin.coroutines.intrinsics.IntrinsicsKt: java.lang.Object getCOROUTINE_SUSPENDED()>();

        if $r1 != $r3 goto label07;

        $r1 = staticinvoke <kotlin.coroutines.intrinsics.IntrinsicsKt: java.lang.Object getCOROUTINE_SUSPENDED()>();

        return $r1;

     label07:
        $r3 = virtualinvoke $r0.<kotlinx.coroutines.internal.ScopeCoroutine: java.lang.Object makeCompletingOnce$kotlinx_coroutines_core(java.lang.Object)>($r1);

        $r8 = <kotlinx.coroutines.JobSupportKt: kotlinx.coroutines.internal.Symbol COMPLETING_WAITING_CHILDREN>;

        if $r3 != $r8 goto label08;

        $r1 = staticinvoke <kotlin.coroutines.intrinsics.IntrinsicsKt: java.lang.Object getCOROUTINE_SUSPENDED()>();

        return $r1;

     label08:
        $z1 = $r3 instanceof kotlinx.coroutines.CompletedExceptionally;

        if $z1 == 0 goto label16;

        $r7 = (kotlinx.coroutines.CompletedExceptionally) $r3;

        $r5 = $r7.<kotlinx.coroutines.CompletedExceptionally: java.lang.Throwable cause>;

        $z1 = $r5 instanceof kotlinx.coroutines.TimeoutCancellationException;

        if $z1 == 0 goto label09;

        $r9 = (kotlinx.coroutines.TimeoutCancellationException) $r5;

        $r10 = $r9.<kotlinx.coroutines.TimeoutCancellationException: kotlinx.coroutines.Job coroutine>;

        if $r10 == $r0 goto label10;

     label09:
        $z0 = 1;

     label10:
        if $z0 == 0 goto label13;

        $r5 = $r7.<kotlinx.coroutines.CompletedExceptionally: java.lang.Throwable cause>;

        $r4 = $r0.<kotlinx.coroutines.internal.ScopeCoroutine: kotlin.coroutines.Continuation uCont>;

        $z0 = staticinvoke <kotlinx.coroutines.DebugKt: boolean getRECOVER_STACK_TRACES()>();

        if $z0 == 0 goto label12;

        $z0 = $r4 instanceof kotlin.coroutines.jvm.internal.CoroutineStackFrame;

        if $z0 != 0 goto label11;

        goto label12;

     label11:
        $r11 = (kotlin.coroutines.jvm.internal.CoroutineStackFrame) $r4;

        $r5 = staticinvoke <kotlinx.coroutines.internal.StackTraceRecoveryKt: java.lang.Throwable access$recoverFromStackFrame(java.lang.Throwable,kotlin.coroutines.jvm.internal.CoroutineStackFrame)>($r5, $r11);

     label12:
        throw $r5;

     label13:
        $z0 = $r1 instanceof kotlinx.coroutines.CompletedExceptionally;

        if $z0 == 0 goto label17;

        $r7 = (kotlinx.coroutines.CompletedExceptionally) $r1;

        $r5 = $r7.<kotlinx.coroutines.CompletedExceptionally: java.lang.Throwable cause>;

        $r4 = $r0.<kotlinx.coroutines.internal.ScopeCoroutine: kotlin.coroutines.Continuation uCont>;

        $z0 = staticinvoke <kotlinx.coroutines.DebugKt: boolean getRECOVER_STACK_TRACES()>();

        if $z0 == 0 goto label15;

        $z0 = $r4 instanceof kotlin.coroutines.jvm.internal.CoroutineStackFrame;

        if $z0 != 0 goto label14;

        goto label15;

     label14:
        $r11 = (kotlin.coroutines.jvm.internal.CoroutineStackFrame) $r4;

        $r5 = staticinvoke <kotlinx.coroutines.internal.StackTraceRecoveryKt: java.lang.Throwable access$recoverFromStackFrame(java.lang.Throwable,kotlin.coroutines.jvm.internal.CoroutineStackFrame)>($r5, $r11);

     label15:
        throw $r5;

     label16:
        $r1 = staticinvoke <kotlinx.coroutines.JobSupportKt: java.lang.Object unboxState(java.lang.Object)>($r3);

     label17:
        return $r1;

        catch java.lang.Throwable from label01 to label02 with label03;
        catch java.lang.Throwable from label04 to label05 with label03;
    }

    private static final java.lang.Object undispatchedResult(kotlinx.coroutines.internal.ScopeCoroutine, kotlin.jvm.functions.Function1, kotlin.jvm.functions.Function0)
    {
        kotlinx.coroutines.internal.ScopeCoroutine $r0;
        kotlin.jvm.functions.Function1 $r1;
        kotlin.jvm.functions.Function0 $r2;
        java.lang.Object $r3, $r6;
        java.lang.Throwable $r4;
        kotlinx.coroutines.CompletedExceptionally $r5;
        kotlinx.coroutines.internal.Symbol $r7;
        boolean $z0;
        java.lang.Boolean $r8;
        kotlin.coroutines.Continuation $r9;
        kotlin.coroutines.jvm.internal.CoroutineStackFrame $r10;

        $r0 := @parameter0: kotlinx.coroutines.internal.ScopeCoroutine;

        $r1 := @parameter1: kotlin.jvm.functions.Function1;

        $r2 := @parameter2: kotlin.jvm.functions.Function0;

     label01:
        $r3 = interfaceinvoke $r2.<kotlin.jvm.functions.Function0: java.lang.Object invoke()>();

     label02:
        goto label04;

     label03:
        $r4 := @caughtexception;

        $r5 = new kotlinx.coroutines.CompletedExceptionally;

        specialinvoke $r5.<kotlinx.coroutines.CompletedExceptionally: void <init>(java.lang.Throwable,boolean,int,kotlin.jvm.internal.DefaultConstructorMarker)>($r4, 0, 2, null);

        $r3 = $r5;

     label04:
        $r6 = staticinvoke <kotlin.coroutines.intrinsics.IntrinsicsKt: java.lang.Object getCOROUTINE_SUSPENDED()>();

        if $r3 != $r6 goto label05;

        $r3 = staticinvoke <kotlin.coroutines.intrinsics.IntrinsicsKt: java.lang.Object getCOROUTINE_SUSPENDED()>();

        return $r3;

     label05:
        $r6 = virtualinvoke $r0.<kotlinx.coroutines.internal.ScopeCoroutine: java.lang.Object makeCompletingOnce$kotlinx_coroutines_core(java.lang.Object)>($r3);

        $r7 = <kotlinx.coroutines.JobSupportKt: kotlinx.coroutines.internal.Symbol COMPLETING_WAITING_CHILDREN>;

        if $r6 != $r7 goto label06;

        $r3 = staticinvoke <kotlin.coroutines.intrinsics.IntrinsicsKt: java.lang.Object getCOROUTINE_SUSPENDED()>();

        return $r3;

     label06:
        $z0 = $r6 instanceof kotlinx.coroutines.CompletedExceptionally;

        if $z0 == 0 goto label12;

        $r5 = (kotlinx.coroutines.CompletedExceptionally) $r6;

        $r4 = $r5.<kotlinx.coroutines.CompletedExceptionally: java.lang.Throwable cause>;

        $r6 = interfaceinvoke $r1.<kotlin.jvm.functions.Function1: java.lang.Object invoke(java.lang.Object)>($r4);

        $r8 = (java.lang.Boolean) $r6;

        $z0 = virtualinvoke $r8.<java.lang.Boolean: boolean booleanValue()>();

        if $z0 == 0 goto label09;

        $r4 = $r5.<kotlinx.coroutines.CompletedExceptionally: java.lang.Throwable cause>;

        $r9 = $r0.<kotlinx.coroutines.internal.ScopeCoroutine: kotlin.coroutines.Continuation uCont>;

        $z0 = staticinvoke <kotlinx.coroutines.DebugKt: boolean getRECOVER_STACK_TRACES()>();

        if $z0 == 0 goto label08;

        $z0 = $r9 instanceof kotlin.coroutines.jvm.internal.CoroutineStackFrame;

        if $z0 != 0 goto label07;

        goto label08;

     label07:
        $r10 = (kotlin.coroutines.jvm.internal.CoroutineStackFrame) $r9;

        $r4 = staticinvoke <kotlinx.coroutines.internal.StackTraceRecoveryKt: java.lang.Throwable access$recoverFromStackFrame(java.lang.Throwable,kotlin.coroutines.jvm.internal.CoroutineStackFrame)>($r4, $r10);

     label08:
        throw $r4;

     label09:
        $z0 = $r3 instanceof kotlinx.coroutines.CompletedExceptionally;

        if $z0 == 0 goto label13;

        $r5 = (kotlinx.coroutines.CompletedExceptionally) $r3;

        $r4 = $r5.<kotlinx.coroutines.CompletedExceptionally: java.lang.Throwable cause>;

        $r9 = $r0.<kotlinx.coroutines.internal.ScopeCoroutine: kotlin.coroutines.Continuation uCont>;

        $z0 = staticinvoke <kotlinx.coroutines.DebugKt: boolean getRECOVER_STACK_TRACES()>();

        if $z0 == 0 goto label11;

        $z0 = $r9 instanceof kotlin.coroutines.jvm.internal.CoroutineStackFrame;

        if $z0 != 0 goto label10;

        goto label11;

     label10:
        $r10 = (kotlin.coroutines.jvm.internal.CoroutineStackFrame) $r9;

        $r4 = staticinvoke <kotlinx.coroutines.internal.StackTraceRecoveryKt: java.lang.Throwable access$recoverFromStackFrame(java.lang.Throwable,kotlin.coroutines.jvm.internal.CoroutineStackFrame)>($r4, $r10);

     label11:
        throw $r4;

     label12:
        $r3 = staticinvoke <kotlinx.coroutines.JobSupportKt: java.lang.Object unboxState(java.lang.Object)>($r6);

     label13:
        return $r3;

        catch java.lang.Throwable from label01 to label02 with label03;
    }
}
